src/relentless/model/potential/bond.py

```diff
@@ -0,0 +1,568 @@
+
+        if s:
+            d = d.item()
+        return d
+
+
+class FENEWCA(BondPotential):
+    r"""Finitely extensible nonlinear elastic (FENE) + Weeks-Chandler-Andersen
+    (WCA) bond interaction.
+
+    .. math::
+
+        u(r) = - \frac{1}{2} k r_0^2 \ln \left[ 1 - \left( \frac{r}{r_0}
+                \right)^2 \right] + u_{\rm{WCA}}(r)
+
+    where
+
+    .. math::
+        u_{\rm{WCA}}(r)  =
+        \begin{cases} 4 \varepsilon \left[ \left( \frac{\sigma}{r}
+                                \right)^{12} - \left( \frac{\sigma}{r}
+                                \right)^{6} \right]  + \varepsilon
+                               & r < 2^{\frac{1}{6}}\sigma\\
+        0 & r \ge 2^{\frac{1}{6}}\sigma
+        \end{cases}
+
+    where :math:`r` is the distance between two bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+-----------+
+    | Parameter   | Description                                      | Initial   |
+    +=============+==================================================+===========+
+    | ``k``       | Spring constant :math:`k`.                       |           |
+    +-------------+--------------------------------------------------+-----------+
+    | ``r0``      | Minimum-energy length :math:`r_0`.               |           |
+    +-------------+--------------------------------------------------+-----------+
+    | ``epsilon`` | Interaction energy :math:`\varepsilon`.          |     0     |
+    +-------------+--------------------------------------------------+-----------+
+    | ``sigma``   | Interaction length :math:`\sigma`.               |     0     |
+    +-------------+--------------------------------------------------+-----------+
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`BondParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Kremer-Grest bond (no repulsion included)::
+
+        >>> u = relentless.potential.bond.FENEWCA(("A"))
+        >>> u.coeff["A"].update({'k': 30, 'r0': 1.5})
+
+    Kremer-Grest bond (repulsion included)::
+
+        >>> u = relentless.potential.bond.FENEWCA(("A"))
+        >>> u.coeff["A"].update({'k': 30, 'r0': 1.5, "epsilon":1, "sigma": 1})
+    """
+
+    def __init__(self, types, name=None):
+        super().__init__(keys=types, params=("k", "r0", "epsilon", "sigma"), name=name)
+
+    def energy(self, types, r):
+        """Evaluate bond energy."""
+        params = self.coeff.evaluate(types)
+        k = params["k"]
+        r0 = params["r0"]
+        epsilon = params["epsilon"]
+        sigma = params["sigma"]
+
+        # initialize arrays
+        r, FENE, s = self._zeros(r)
+        WCA = FENE.copy()
```

Comment:
Prefer not to use all capitals for variables, as sometimes these indicate constants. This would be better:
```suggestion
        r, u_fene, s = self._zeros(r)
        u_wca = u_fene.copy()
```